src/main/java/hudson/remoting/Checksum.java

```diff
@@ -74,8 +74,10 @@ static Checksum forFile(File file) throws IOException {
     static Checksum forURL(URL url) throws IOException {
         try {
             MessageDigest md = MessageDigest.getInstance(JarLoaderImpl.DIGEST_ALGORITHM);
-            Util.copy(url.openStream(), new DigestOutputStream(new NullOutputStream(), md));
-            return new Checksum(md.digest(), md.getDigestLength() / 8);
+            try(OutputStream ostream = new DigestOutputStream(new NullOutputStream(), md)) {
```

Comment:
If you actually were concerned about `NullOutputStream` leaking something (impossible), you would use

```java
try (NullOutputStream nos = new NullOutputStream();
     OutputStream ostream = new DigestOutputStream(nos)) {
```